DQMOffline/Lumi/src/ElectronIdentifier.cc

```diff
@@ -0,0 +1,174 @@
+   cuts_[EleIDCutNames::MISSINGHITS][EleIDWorkingPoints::LOOSE] [EleIDEtaBins::ENDCAP] = 1;
+   cuts_[EleIDCutNames::MISSINGHITS][EleIDWorkingPoints::MEDIUM][EleIDEtaBins::ENDCAP] = 1;
+   cuts_[EleIDCutNames::MISSINGHITS][EleIDWorkingPoints::TIGHT] [EleIDEtaBins::ENDCAP] = 1;
+
+   cuts_[EleIDCutNames::CONVERSION][EleIDWorkingPoints::VETO]  [EleIDEtaBins::BARREL] = 1;
+   cuts_[EleIDCutNames::CONVERSION][EleIDWorkingPoints::LOOSE] [EleIDEtaBins::BARREL] = 1;
+   cuts_[EleIDCutNames::CONVERSION][EleIDWorkingPoints::MEDIUM][EleIDEtaBins::BARREL] = 1;
+   cuts_[EleIDCutNames::CONVERSION][EleIDWorkingPoints::TIGHT] [EleIDEtaBins::BARREL] = 1;
+
+   cuts_[EleIDCutNames::CONVERSION][EleIDWorkingPoints::VETO]  [EleIDEtaBins::ENDCAP] = 1;
+   cuts_[EleIDCutNames::CONVERSION][EleIDWorkingPoints::LOOSE] [EleIDEtaBins::ENDCAP] = 1;
+   cuts_[EleIDCutNames::CONVERSION][EleIDWorkingPoints::MEDIUM][EleIDEtaBins::ENDCAP] = 1;
+   cuts_[EleIDCutNames::CONVERSION][EleIDWorkingPoints::TIGHT] [EleIDEtaBins::ENDCAP] = 1;
+}
+
+void ElectronIdentifier::setRho(double rho) {
+   if(rho >= 0) {
+      rho_ = rho;
+   } else {
+      throw cms::Exception("ValueError")
+         << "Encountered invalid value for energy density rho.\n"
+         << "Value: " << rho << "\n"
+         << "Rho should be a real, positive number.\n";
+   }
+}
+void ElectronIdentifier::setID(std::string ID) {
+   if(ID=="TIGHT") ID_ = EleIDWorkingPoints::TIGHT;
+   else if(ID=="MEDIUM") ID_ = EleIDWorkingPoints::MEDIUM;
+   else if(ID=="LOOSE") ID_ = EleIDWorkingPoints::LOOSE;
+   else if(ID=="VETO") ID_ = EleIDWorkingPoints::VETO;
+   else throw;
+}
+float ElectronIdentifier::dEtaInSeed(const reco::GsfElectronPtr& ele){
+         return ele->superCluster().isNonnull() && ele->superCluster()->seed().isNonnull() ?
+         ele->deltaEtaSuperClusterTrackAtVtx() - ele->superCluster()->eta() + ele->superCluster()->seed()->eta() : std::numeric_limits<float>::max();
+      }
+float ElectronIdentifier::isolation(const reco::GsfElectronPtr& ele) {
+  if(rho_ < 0 ) {
+     throw;
+  }
+  const reco::GsfElectron::PflowIsolationVariables& pfIso = ele->pfIsolationVariables();
+  const float chad = pfIso.sumChargedHadronPt;
+  const float nhad = pfIso.sumNeutralHadronEt;
+  const float pho = pfIso.sumPhotonEt;
+  const float  eA = _effectiveAreas.getEffectiveArea( fabs(ele->superCluster()->eta()) );
+  const float iso = chad + std::max(0.0, nhad + pho - rho_*eA);
+
+  // Apply the cut and return the result
+  // Scale by pT if the relative isolation is requested but avoid division by 0
+  return iso;
+}
+
+
+bool ElectronIdentifier::passID(const reco::GsfElectronPtr& ele,edm::Handle<reco::BeamSpot> beamspot,edm::Handle<reco::ConversionCollection> conversions) {
+   if(ID_ == -1) throw;
+   unsigned int region = fabs(ele->superCluster()->eta()) < 1.479 ? EleIDEtaBins::BARREL : EleIDEtaBins::BARREL;
+
+   bool pass = true;
+
+   std::vector<bool> passes;
```

Comment:
this can become a list of if statements that exit if false, right? Once one of these is false, there is no reason to continue through the evaluation that I see.

Reply:
You are correct. Switched to list of ifs.